src/jwt_authorizer/analyze.py

```diff
@@ -0,0 +1,80 @@
+"""Token and ticket analysis.
+
+Analyze tokens and tickets and display their contents and other debugging
+information.
+"""
+
+from __future__ import annotations
+
+import base64
+from typing import TYPE_CHECKING
+
+import jwt
+
+from jwt_authorizer.tokens import ALGORITHM
+
+if TYPE_CHECKING:
+    from jwt_authorizer.tokens import Ticket, TokenStore, TokenVerifier
+    from typing import Any, Dict
+
+
+def analyze_ticket(
+    ticket: Ticket, token_store: TokenStore, token_verifier: TokenVerifier
+) -> Dict[str, Any]:
+    """Analyze a ticket and return its expanded information.
+
+    Parameters
+    ----------
+    ticket : `Ticket`
+        The parsed ticket to analyze.
+    token_store : `TokenStore`
+        The backend store used to retrieve the session for the ticket.
+    token_verifier : `TokenVerifier`
+        Verifier to check the validity of any underlying token.
+
+    Returns
+    -------
+    output : `Dict` [`str`, `Any`]
+        The contents of the ticket.  This will include the ticket ID and
+        secret, the session it references, and the token that session
+        contains.
+    """
+    output: Dict[str, Any] = {
+        "ticket": {
+            "ticket_id": ticket.ticket_id,
+            "secret": base64.urlsafe_b64encode(ticket.secret).decode(),
+        }
+    }
+
+    session = token_store.get_session(ticket)
+    if not session:
+        output["errors"] = ["No session found for {ticket.as_handle()}"]
```

Comment:
Does this need an `f` prefix or is it interpreted somewhere else?

Reply:
Good catch, thanks!